src/lib/history.ts

```diff
@@ -0,0 +1,29 @@
+import { WeeklyArticleLog, WeeklyArticleHistory } from '../components/ContributionsEpicTypes';
+
+export const getMondayFromDate = (date: Date): number => {
+    const day = date.getDay() || 7;
+    if (day !== 1) {
+        date.setHours(-24 * (day - 1));
+    }
+    return Math.floor(date.getTime() / 86400000);
+};
+
+export const getArticleViewCountForWeeks = (
+    history: WeeklyArticleHistory = [],
+    weeks: number,
+): number => {
+    const mondayThisWeek = getMondayFromDate(new Date());
+    const cutOff = mondayThisWeek - weeks * 7;
+
+    const firstOldWeekIndex = history.findIndex(
+        (c: WeeklyArticleLog) => c.week && c.week <= cutOff,
+    );
+
+    const articleCountWindow =
+        firstOldWeekIndex >= 0 ? history.slice(0, firstOldWeekIndex) : history;
```

Comment:
I wonder if we could combine these two into one `filter` call? Since logically aren't we filtering out all the old weeks, so something like:

```js
const articleCountWindow = history.filter(c => c.week > cutOff);
```

Not sure that's exactly right, but something similar? We could even give the filter a name if it helped readability:

```js
const weekWithinCutoff = c => c.week > cutOff;
const articleCountWindow = history.filter(weekWithinCutoff);
```

Reply:
Yeah that makes sense, I've refactored as suggested.